lib/msf/core/payload/linux/x64/reverse_tcp.rb

```diff
@@ -161,8 +166,9 @@ def asm_reverse_tcp(opts={})
 
       recv:
         pop    rsi
+        push   0x#{read_length.to_s(16)}
```

Comment:
If we add this here we should also remove the previous pushed size at https://github.com/rapid7/metasploit-framework/blob/97943261ed53584f982cb37cff62665e7d22a9dd/lib/msf/core/payload/linux/x64/reverse_tcp.rb#L115